observer/observer.go

```diff
@@ -169,14 +191,16 @@ func (o *Observer) createOrReplaceMessagesView(ctx context.Context, includeParqu
 }
 
 func (o *Observer) observe(ctx context.Context) error {
-	rotation := time.NewTimer(o.rotateInterval)
+	rotation := time.NewTicker(o.rotateInterval)
+	flush := time.NewTicker(o.maxBatchDelay)
```

Comment:
Another note, this will lead to a maximum batch time of 2x maxBatchDelay, but I'm fine with that.